src/metrics/mongoose_metrics_hooks.erl

```diff
@@ -16,89 +16,91 @@
-      {roster_push, HostType, ?MODULE, roster_push, 50},
-      {roster_in_subscription, HostType, ?MODULE, roster_in_subscription, 55},
-      {register_user, HostType, ?MODULE, register_user, 50},
-      {remove_user, HostType, ?MODULE, remove_user, 50},
-      {privacy_iq_get, HostType, ?MODULE, privacy_iq_get, 1},
-      {privacy_iq_set, HostType, ?MODULE, privacy_iq_set, 1},
-      {privacy_check_packet, HostType, ?MODULE, privacy_check_packet, 55},
-      {sm_broadcast, HostType, ?MODULE, privacy_list_push, 1}].
+    [ {sm_register_connection_hook, HostType, fun ?MODULE:sm_register_connection_hook/3, #{}, 50},
+      {sm_remove_connection_hook, HostType, fun ?MODULE:sm_remove_connection_hook/3, #{}, 50},
+      {auth_failed, HostType, fun ?MODULE:auth_failed/3, #{}, 50},
+      {user_send_packet, HostType, fun ?MODULE:user_send_packet/3, #{}, 50},
+      {user_receive_packet, HostType, fun ?MODULE:user_receive_packet/3, #{}, 50},
+      {xmpp_stanza_dropped, HostType, fun ?MODULE:xmpp_stanza_dropped/3, #{}, 50},
+      {xmpp_bounce_message, HostType, fun ?MODULE:xmpp_bounce_message/3, #{}, 50},
+      {xmpp_send_element, HostType, fun ?MODULE:xmpp_send_element/3, #{}, 50},
+      {roster_get, HostType, fun ?MODULE:roster_get/3, #{}, 55},
+      {roster_set, HostType, fun ?MODULE:roster_set/3, #{}, 50},
+      {roster_push, HostType, fun ?MODULE:roster_push/3, #{}, 50},
+      {roster_in_subscription, HostType, fun ?MODULE:roster_in_subscription/3, #{}, 55},
+      {register_user, HostType, fun ?MODULE:register_user/3, #{}, 50},
+      {remove_user, HostType, fun ?MODULE:remove_user/3, #{}, 50},
+      {privacy_iq_get, HostType, fun ?MODULE:privacy_iq_get/3, #{}, 1},
+      {privacy_iq_set, HostType, fun ?MODULE:privacy_iq_set/3, #{}, 1},
+      {privacy_check_packet, HostType, fun ?MODULE:privacy_check_packet/3, #{}, 55},
+      {sm_broadcast, HostType, fun ?MODULE:privacy_list_push/3, #{}, 1}].
 
--spec sm_register_connection_hook(any(), mongooseim:host_type(), tuple(), jid:jid(), term()
-                                 ) -> any().
-sm_register_connection_hook(Acc, HostType, _, _, _) ->
+-spec sm_register_connection_hook(Acc, Params, Extra) -> {ok, Acc} when
+      Acc :: any(),
+      Params :: map(),
+      Extra :: #{host_type := mongooseim:host_type()}.
+sm_register_connection_hook(Acc, _, #{host_type := HostType}) ->
     mongoose_metrics:update(HostType, sessionSuccessfulLogins, 1),
     mongoose_metrics:update(HostType, sessionCount, 1),
     Acc.
 
--spec sm_remove_connection_hook(mongoose_acc:t(), tuple(), jid:jid(),
-                                term(), ejabberd_sm:close_reason()
-                               ) -> mongoose_acc:t().
-sm_remove_connection_hook(Acc, _, _, _, _Reason) ->
+-spec sm_remove_connection_hook(Acc, Params, Extra) -> {ok, Acc} when
+      Acc :: mongoose_acc:t(),
+      Params :: map(),
+      Extra :: map().
+sm_remove_connection_hook(Acc, _, _) ->
     HT = mongoose_acc:host_type(Acc),
```

Comment:
Actually, just like for `sm_register_connection_hook` we're extracting the `host_type` from the `Extra` argument, we can do the same in this one and the many other hooks that extract it ad-hoc from the `mongoose_acc`.

Reply:
done 👍